### ISSUE TITLE
Container description vanishes after migration

### ISSUE THREAD

**Issue by DeadlyMercury** [lxc/lxd#4706]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 18.04
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.13
   * LXC version: 3.0
   * LXD version: 3.0
   * Storage backend in use: zfs

# Issue description
Description field vanish if container moved to another node. I was making some works on my nodes so migrating all containers between nodes. And now I have no descriptions at my containers at all, and that is depressing me :(
I want description held if ct was moved or copied.

Also I don't understand how to set description with command like "lxc config set ct description"... There is no hint for that and description is unknown key. Maybe it could not be done like that?

# Steps to reproduce

 1. Create ct!
`# lxc launch stretch test`
 2. Set description!
`# lxc config edit test`
 3. Read it!
```
# lxc list -c ns4SPd
+--------+---------+------+-----------+----------+--------------------------+
|  NAME  |  STATE  | IPV4 | SNAPSHOTS | PROFILES |       DESCRIPTION        |
+--------+---------+------+-----------+----------+--------------------------+
| bl-adm | STOPPED |      | 0         | default  |                          |
+--------+---------+------+-----------+----------+--------------------------+
| test   | RUNNING |      | 0         | default  | my marvelous description |
+--------+---------+------+-----------+----------+--------------------------+
```
 4. Break it!
```
# lxc stop test && lxc move test node02: && lxc move node02:test test
# lxc list -c ns4SPd
+--------+---------+------+-----------+----------+-------------+
|  NAME  |  STATE  | IPV4 | SNAPSHOTS | PROFILES | DESCRIPTION |
+--------+---------+------+-----------+----------+-------------+
| bl-adm | STOPPED |      | 0         | default  |             |
+--------+---------+------+-----------+----------+-------------+
| test   | STOPPED |      | 0         | default  |             |
+--------+---------+------+-----------+----------+-------------+
```

 _5. Set it again... for all containers on all nodes... via "config edit"... Don't cry!_

**Comment by brauner** [comment]
@stgraber, do you know from the top of your head where this is set?

**Comment by stgraber** [comment]
The description is an attribute of containerArgs, so I suspect there's a place where we don't set that field properly for migrated containers... It's odd because I remember fixing something like this before.

**Comment by DeadlyMercury** [comment]
Will u backport this fix to 3.0 branch?
Also is there any way to set description via "lxc config set..."?

**Comment by stgraber** [comment]
Yeah, it'll be in 3.0.2.

There's no way to set the description through "lxc config set" as it's not a configuration field but a property of the container instead. Your best bet from a shell script is to use `lxc config show NAME | sed ... | lxc config edit NAME`
